### PR TITLE
[usage] Add db.WorkspaceInstance model in golang

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Same as https://github.com/gitpod-io/gitpod/pull/10293 but for WorkspaceInstances.

## Related Issue(s)

* Part of https://github.com/gitpod-io/gitpod/issues/10317
* Depends on https://github.com/gitpod-io/gitpod/pull/10390

## How to test

Unit tests

## Release Notes

```release-note
NONE
```

## Documentation

NONE

### CODE REVIEW

**Comment by roboquat** [comment]
@easyCZ: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/usage/pkg/db/workspace_instance.go
Comment (geropl) [inline components/usage/pkg/db/workspace_instance.go:16]: nit: one ID column is UUID, the other string: Why not make both strings? (so far we only have one column type for IDs: string) What's the advantange of UUID?

**Review by geropl — APPROVED** [review]
code looks good, tests are green! :heavy_check_mark: 

/hold because of [this question](https://github.com/gitpod-io/gitpod/pull/10367/files#r886544711). I leave it up to you to handle. :pray:

File: components/usage/pkg/db/workspace_instance.go
Comment (easyCZ) [reply components/usage/pkg/db/workspace_instance.go:16]: UUID allows you to parse UUIDs and validate before you even hit the DB. UUID is the more appropriate type as it's more specific. For example, it prevents you from inserting something like `Workspace{ID: "my-random-string"}` because it won't even compile

Comment (easyCZ) [reply components/usage/pkg/db/workspace_instance.go:16]: And it is. `uuid.UUID` serializes as a string. It's the same as using the UUID_COLUMN type in our TypeORM implementation. 

https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/src/typeorm/entity/db-workspace-instance.ts#L24

Comment (geropl) [reply components/usage/pkg/db/workspace_instance.go:16]: Ah! The UUID_COLUMN_TYPE is only there to ensure the width/type of the field is the same everywhere.

The key question is: does `uuid.UUID` influence the runtime behavior (e.g., introduces new error modes during parsing? If no I'm happy! :+1:

Comment (easyCZ) [reply components/usage/pkg/db/workspace_instance.go:16]: It shouldn't, as long as what is stored in the DB is a valid UUID. I'm gonna run it against the RO replica of staging & prod to batch list all items to validate we don't have underlying violations to verify.

**Pull request merged**
